src/node_main_instance.cc

```diff
@@ -80,6 +84,14 @@ NodeMainInstance::NodeMainInstance(const SnapshotData* snapshot_data,
       isolate_params_.get(), event_loop, platform, snapshot_data != nullptr);
   CHECK_NOT_NULL(isolate_);
 
+  cpp_heap_ = CppHeap::Create(
+      platform,
+      CppHeapCreateParams{{},
+                          WrapperDescriptor(BaseObject::kEmbedderType,
+                                            BaseObject::kSlot,
+                                            kNodeEmbedderIdForCppgc)});
+  isolate_->AttachCppHeap(cpp_heap_.get());
```

Comment:
From the doc looks like this can only be done once per isolate. @codebytere would doing this unconditionally be in conflict with what Electron does? (From https://github.com/nodejs/node/pull/43521 it seems that this would be called by someone else in Electron too).

Reply:
This PR only touches `CommonEnvironmentSetup`, `NodeMainInstance` and `WorkerThreadData`, so if Electron isn't using any of these (from what I can tell it's using `CreateEnvironment()` and `LoadEnvironment()` now which should be unaffected by this PR) I think it should be fine.